File: java-to-kotlin/src/main/java/nl/rabobank/kotlinmovement/recipes/model/RecipeRequest.java
package nl.rabobank.kotlinmovement.recipes.model;


import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import lombok.Data;

import java.util.Set;

@Data
public final class RecipeRequest {
    @NotBlank( message = "recipeName")
    final String recipeName;
    @NotEmpty(message = "ingredients")
    @Valid
    private final Set<IngredientRequest> ingredients;
}

File: non-blocking-complete/src/main/kotlin/nl/alimeshkat/recipes/model/RecipeRequest.kt
package nl.alimeshkat.recipes.model

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty

data class RecipeRequest(
    @field:NotBlank(message = "recipeName")
    val recipeName: String?,
    @field:NotEmpty(message = "ingredients") @field:Valid
    val ingredients: MutableSet<IngredientRequest>?
)
